tests/performance-test/performance-test.sh

```diff
@@ -49,91 +49,170 @@ ellipse(){
-        exit 1
-    fi
-    OCP_PROM_HOST="https:\/\/$(oc get routes --field-selector metadata.name=prometheus-k8s -o jsonpath="{.items[0].spec.host}")"
-    OCP_DATASOURCE=$(oc get secret -n openshift-monitoring grafana-datasources -o jsonpath='{.data.prometheus\.yaml}' \
-        | base64 -d)
-    OCP_DATASOURCE=$(echo "${OCP_DATASOURCE//$'\n'/}" | sed 's/.*\[\([^]]*\)\].*/\1/g') #get DS json from between brackets
-    OCP_DATASOURCE=$(echo "$OCP_DATASOURCE" | sed 's/"name": "[a-z]*"/"name": "OCPPrometheus"/g')  #Change DS name
-    OCP_DATASOURCE=$(echo "$OCP_DATASOURCE" | sed 's/"url": "[^,]*"/"url": "PROMHOST"/g')     #DS placeholder url
-    OCP_DATASOURCE=$(echo "$OCP_DATASOURCE" | sed "s/\"url\": \"PROMHOST\"/\"url\": \"${OCP_PROM_HOST}\"/g") #Change DS url
-
-    if ! oc project sa-telemetry; then 
+# basic check if necessary resources exist. DOES NOT verify they work correctly
+check_resources(){
+    printf "\n*** Performing Resource Checks ***\n"
+    if ! oc get project sa-telemetry; then 
         echo "Error: SAF does not exist on cluster. Deploy SAF before running performance test" 1>&2
         exit 1
     fi
+
+    if ! oc get project openshift-monitoring; then 
+        echo "Error: openshift-monitoring not enabled on host. Continuing without it will result \
+        in reduced functionality of performance test. Would you like to continue without \
+        it? [y/n]"
+        read RESP
+        case $RESP in
+            "y") 
+            ;;
+            *)
+                exit 1
+            ;;
+        esac
+    fi
+
     if ! GRAF_HOST=$(oc get routes --field-selector metadata.name=grafana -o jsonpath="{.items[0].spec.host}") 2> /dev/null; then 
         echo "Error: cannot find Grafana instance in cluster. Has it been deployed?" 1>&2
         exit 1
     fi
 
-    SAF_PROM_HOST=$(oc get routes --field-selector metadata.name=prometheus -o jsonpath="{.items[0].spec.host}")
+    local datasources=$(curl --silent -X GET "http://${GRAF_HOST}/api/datasources")
+    if ! echo $datasources | grep -q "OCPPrometheus"; then
+        echo "Error: unable to find Grafana datasource OCPPrometheus"
+        exit 1
+    fi
 
-    SAF_DATASOURCE=$(cat << EOF
-        {
-            "name"  :   "SAFPrometheus",
-            "type"  :   "prometheus",
-            "url"   :   "http://$SAF_PROM_HOST",
-            "access":   "direct",
-            "basicAuth" :false,
-            "jsonData": {
-                "timeInterval"  :   "1s"
-            }
-        }
-EOF
-)
+    if ! echo $datasources | grep -q "SAFPrometheus"; then
+        echo "Error: unable to find Grafana datasource SAFPrometheus"
+        exit 1
+    fi
+
+    if ! echo $datasources | grep -q "SAFElasticsearch"; then
+        echo "Error: unable to find Grafana datasource SAFElasticsearch"
+        exit 1
+    fi
 }
 
+# Post dashboards to grafana - overwrites if they already exist
```

Comment:
Just a comment that I actually nerf this every time because I tend to tweak dashboards as I'm testing. An option to disable it would be nice.

Reply:
I should add datasources to the Grafana provisioning stage anyways - then it will only be written once. Future PR

Reply:
Don't forget to open a low priority issue so you don't forget! :)

Reply:
Good point!